src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        copyState = { ...copyState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        copyState = { ...copyState };
+
+        for (const key of action.keysToRemove) {
+          delete copyState[key];
+        }
+        break;
+
+      case 'clear':
+        copyState = {};
```

Comment:
The 'clear' case is missing a 'break' statement. Without it, the function will continue executing the next case if any are added in the future, which could lead to unintended behavior.